tests/test-extensions/transactional-extension/src/main/java/org/mule/test/transactional/TransactionalSource.java

```diff
@@ -81,6 +81,11 @@ public void onStart(SourceCallback<TestTransactionalConnection, Object> sourceCa
             txHandle.commit();
           }
         } catch (TransactionException e) {
+          try {
+            txHandle.rollback();
+          } catch (TransactionException e1) {
+            throw new RuntimeException(e1);
```

Comment:
I don't think this is correct. If the rollback fails you should log that but the exception raised should still be the original one. Otherwise you're hiding the root cause